src/main/java/edu/cornell/kfs/coa/batch/service/impl/CopyLegacyAccountAttachmentsServiceImpl.java

```diff
@@ -0,0 +1,202 @@
+                    legacyAccountAttachment.getKfsAccountNumber(), e);
+            return false;
+        }
+    }
+
+    private void saveAttachmentToKfs(final LegacyAccountAttachment legacyAccountAttachment,
+            final DataBuffer fileContents) throws IOException {
+        final Account account = accountService.getByPrimaryId(legacyAccountAttachment.getKfsChartCode(),
+                legacyAccountAttachment.getKfsAccountNumber());
+        if (ObjectUtils.isNull(account)) {
+            throw new RuntimeException("Account '" + legacyAccountAttachment.getKfsChartCode()
+                    + "-" + legacyAccountAttachment.getKfsAccountNumber() + "' does not exist");
+        }
+        final Person systemUser = personService.getPersonByPrincipalName(KFSConstants.SYSTEM_USER);
+        final String noteText = createNoteText(legacyAccountAttachment);
+        final Attachment attachment = createKfsAttachment(legacyAccountAttachment, fileContents, account);
+
+        final Note note = new Note();
+        note.setAuthorUniversalIdentifier(systemUser.getPrincipalId());
+        note.setNoteTypeCode(NoteType.BUSINESS_OBJECT.getCode());
+        note.setRemoteObjectIdentifier(account.getObjectId());
+        note.setNoteText(noteText);
+        note.setNotePostedTimestampToCurrent();
+        note.setAttachment(attachment);
+        noteService.save(note);
+    }
+
+    private Attachment createKfsAttachment(final LegacyAccountAttachment legacyAccountAttachment,
+            final DataBuffer fileContents, final Account account) throws IOException {
+        final String fileName = legacyAccountAttachment.getFileName();
+        final String mimeType = CuAttachmentUtils.findMimeType(fileName);
+        final int fileSize = fileContents.readableByteCount();
+        try (final InputStream fileStream = fileContents.asInputStream()) {
+            return attachmentService.createAttachment(account, fileName, mimeType, fileSize, fileStream, null);
+        }
+    }
+
+    private String createNoteText(final LegacyAccountAttachment legacyAccountAttachment) {
+        final int noteTextMaxLength = dataDictionaryService.getAttributeMaxLength(Note.class,
+                KRADConstants.NOTE_TEXT_PROPERTY_NAME);
+        final String noteTemplate = configurationService.getPropertyValueAsString(
+                CuCOAKeyConstants.LEGACY_ACCOUNT_ATTACHMENT_NOTE_TEXT_TEMPLATE);
+        final String noteText = String.format(Locale.US, noteTemplate, legacyAccountAttachment.getLegacyAccountCode(),
+                legacyAccountAttachment.getAddedBy(), legacyAccountAttachment.getFileDescription());
+        return StringUtils.left(noteText, noteTextMaxLength);
+    }
+
+    private int getParameterValueAsInteger(final String parameterName) {
+        final String parameterValue = parameterService.getParameterValueAsString(
+                CopyLegacyAccountAttachmentsStep.class, parameterName);
+        return Integer.parseInt(parameterValue);
```

Comment:
Would it be helpful to log the parameter values the first time they are retrieved for the job

Reply:
I'm not sure if logging these values would do much for this job, given that only a couple are needed for the job and the value of one of them is likely to be reflected in the retrieved-batch-size log entries. If you think they would still add significant value, I could add them.